Fix Block Cache reporting in zeebe dashboard.

rodrigo-lourenco-lopes opened the pull request:
## Description

This PR starts displaying the block cash per broker instead of per partition. Additionally, in the RocksDB memory usage we dont count the block cache maximum size as part of memory usage for this panel, adjusting the description accordingly. 

The legacy way of displaying the metrics will still work for versions < 8.8 because we use different queries conditionally on the existance of the memory allocation strategy metric.

This affects the panels `RocksDb Memory Usage`, `Block Cache Capacity`, `Block Cache Usage`, and `Pinned Block Cache Usage`.
<img width="1158" height="444" alt="image" src="https://github.com/user-attachments/assets/ba0a1a2c-a7bf-445c-a427-0a8b6b20ab8c" />

<img width="579" height="420" alt="image" src="https://github.com/user-attachments/assets/7510196f-1e7d-4ad5-83c8-5d9cb7a8b2aa" />


This Pr also adds a panel for displaying the allocation strategy per broker.

<img width="312" height="257" alt="image" src="https://github.com/user-attachments/assets/733c453d-7580-40e5-85bf-c58273446883" />


Aditionally we add a short description that the metrics above might be incorrect for versions lower than 8.8.9.
<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes https://github.com/camunda/camunda/issues/43237

rodrigo-lourenco-lopes committed:
feat: fix rocks db panels in the zeebe dashboard.

This commit starts displaying the block cash per broker instead of per partition. Additionally, in the RocksDB memory usage we dont count the block cache maximum size as part of memory usage for this panel, adjusting the description accordingly.

rodrigo-lourenco-lopes committed:
fix: remove unused partition tag from labels.

rodrigo-lourenco-lopes committed:
fix: fix descriptions.

entangled90 commented:
The absence of the gauge metric would indicate that the version is old and is not "sharing" memory across all partitions. Would that be ok?

rodrigo-lourenco-lopes committed:
feat: add memory allocation strategy metric.

entangled90 reviewed (changes requested):
I was actually referring to use that metric in the dashboard queries themselves so that we always show the right amount.
Not sure what's the best way to do it, but you should be able to do this with and and or, see [here](https://stackoverflow.com/questions/64204913/promql-if-else-like-expression).

rodrigo-lourenco-lopes committed:
feat: conditional metric display based on alloc strat.

For the RocksDb Memory Usage, Block Cache Capacity, Block Cache Usage, and Pinned Block Cache Usage we use the existance of the allocation strat metric to choose to display between the legacy mode and the new.

rodrigo-lourenco-lopes committed:
feat: add new allocation strategy metric.

rodrigo-lourenco-lopes committed:
fix: move panel of strategy to top of row.

entangled90 reviewed (approved):
Thanks 🚀
